examples/systemcalls/systemcalls.c

```diff
@@ -93,7 +142,65 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
  *
 */
 
-    va_end(args);
+    int out_fd = open(outputfile, O_WRONLY | O_TRUNC | O_CREAT, 0644);
+    if (out_fd == -1){
+    	printf("Error, issue opening output file!\r\n");
+    	va_end(args);
+    	return false;
+    }
+    	
+    int pid; 
+    int status; 
+    pid = fork();
+    
+    if (pid == -1){
+    	printf("Issue forking to create child process\r\n");
+    	va_end(args);
```

Comment:
Missing `close(out_fd)` here and in other return paths.  Probably would be easier/better to set a return status and return in one place after cleanup